packages/react-storage/src/components/StorageBrowser/views/Controls/Table.tsx

```diff
@@ -127,20 +128,35 @@ export type RenderRowItem<T> = (row: T, index: number) => JSX.Element;
 interface TableControlProps<T> {
   data: T[];
   columns: Column<T>[];
+  handleDroppedFiles?: (files: File[]) => void;
   renderHeaderItem: RenderHeaderItem<T>;
   renderRowItem: RenderRowItem<T>;
 }
 
 export function TableControl<U>({
   data,
   columns,
+  handleDroppedFiles,
   renderHeaderItem,
   renderRowItem,
 }: TableControlProps<U>): React.JSX.Element {
   const ariaLabel = 'Table';
 
+  const { dragState, ...dropHandlers } = useDropZone({
+    acceptedFileTypes: [],
+    onDropComplete: ({ acceptedFiles }) =>
+      handleDroppedFiles && handleDroppedFiles(acceptedFiles),
+  });
+
   return (
-    <Table aria-label={ariaLabel} className={BLOCK_NAME}>
+    <Table
+      aria-label={ariaLabel}
+      data-testid="storage-browser-table"
+      className={`${BLOCK_NAME} ${
+        dragState !== 'inactive' ? `${BLOCK_NAME}__dropzone` : ''
+      }`}
+      {...dropHandlers}
```

Comment:
Prefer placing spread `props` above explicit `props` to ensure that values in the spread do not override the explicit values
```suggestion
      {...dropHandlers}
      aria-label={ariaLabel}
      data-testid="storage-browser-table"
      className={`${BLOCK_NAME} ${
        dragState !== 'inactive' ? `${BLOCK_NAME}__dropzone` : ''
      }`}
````